common/djangoapps/student/models.py

```diff
@@ -3116,6 +3122,135 @@ class AccountRecoveryConfiguration(ConfigurationModel):
     )
 
 
+class UserCelebration(TimeStampedModel):
+    """
+    Keeps track of how we've celebrated a user's course progress.
+    This class is for course agnostic celebrations (not specific to a particular enrollment).
+    CourseEnrollmentCelebration is for celebrations that happen separately for each separate course.
+
+    The first_day_of_streak, last_day_of_streak and last_streak_celebration fields are used to
```

Comment:
actually i moved it above the user line but pylint didn't like that I had back to back comment blocks so I reverted the change, not sure if you can clarify what you were looking for

Reply:
Again, minor deal. But I was thinking something like:

```
"""
docstring docstring docstring...
.. no_pii:
"""
user = ...

# The first_day_of_streak and last_day_of_streak...
# ...
first_day_of_streak = models.DateField(default=None, null=True, blank=True)
last_day_of_streak = models.DateField(default=None, null=True, blank=True)
```

Like, imagine if we get a new chunk of fields for some other celebration - I'd imagine each chunk of fields would want their own separate explanations, co-located with the fields, rather than the global class docstring.

Reply:
changed